apps/examples/mediarecorder/mediarecorder_main.cpp

```diff
@@ -234,28 +235,27 @@ class MediaRecorderTest : public MediaRecorderObserverInterface
 			return;
 		}
 
-		mp.setObserver(shared_from_this());
-		mp.setDataSource(std::move(source));
+		if (mp.setObserver(shared_from_this()) == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::setObserver failed" << std::endl;
+			return;
+		}
 
-		if (mp.prepare() == PLAYER_OK) {
-			isPlaying = true;
-			if (mp.start() == PLAYER_OK) {
-				// playback started
-				return;
-			}
+		if (mp.setDataSource(std::move(source)) == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::setDataSource failed" << std::endl;
+			return;
+		}	
 
-			std::cout << "Mediaplayer::start failed" << std::endl;
-			isPlaying = false;
+		if (mp.prepare() == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::prepare failed" << std::endl;
+			return;
+		}
 
+		if (mp.start() == PLAYER_ERROR) {
+			std::cout << "Mediaplayer::start failed" << std::endl;
 			if (mp.unprepare() == PLAYER_ERROR) {
 				std::cout << "Mediaplayer::unprepare failed" << std::endl;
 			}
-		} else {
-			std::cout << "Mediaplayer::prepare failed" << std::endl;
-		}
-
-		if (mp.destroy() == PLAYER_ERROR) {
-			std::cout << "Mediaplayer::destroy failed" << std::endl;
+			return;
```

Comment:
Is it okay to remove mp.destroy() after mp.start() finishes?
Please check this.

Reply:
Perform destory() when play is complete (onPlaybackFinished).
Destory is not needed here.